### PR TITLE
Fix an out of bounds access in ForwardMolSupplier

### PR DESCRIPTION
**Pull request by bp-kelley**
An update to the osx toolchain on conga hit this long standing bug,

### CODE REVIEW

File: Code/GraphMol/FileParsers/ForwardSDMolSupplier.cpp [hunk 1]
```diff
@@ -116,8 +116,8 @@ void ForwardSDMolSupplier::readMolProps(ROMol &mol) {
           std::string prop = "";
           auto stmp = FileParserUtils::strip(tempStr);
           int nplines = 0;  // number of lines for this property
-          while (stmp.length() != 0 || tempStr[0] == ' ' ||
-                 tempStr[0] == '\t') {
+          while (stmp.length() != 0 || (tempStr.length() && (tempStr[0] == ' ' ||
```
Comment (greglandrum) [inline Code/GraphMol/FileParsers/ForwardSDMolSupplier.cpp:119]: Please use .empty() instead of comparing length() with zero

**Review by greglandrum — CHANGES_REQUESTED** [review]
Minor changes

**Comment by greglandrum** [comment]
@bp-kelley The Mac tests are still (again?) failing

**Comment by bp-kelley** [comment]
@greglandrum I fixed the remaining errors.  I also switched from string[0] to string.at(0) to help me find the issues.  We can switch this back if you like.

**Comment by bp-kelley** [comment]
@greglandrum yep, it's fixed

**Review by greglandrum — APPROVED** [review]
LGTM

**Pull request merged**